pandas/core/arrays/string_arrow.py

```diff
@@ -36,14 +36,20 @@
     except ImportError:
         pass
     else:
-        ARROW_CMP_FUNCS = {
-            "eq": pc.equal,
-            "ne": pc.not_equal,
-            "lt": pc.less,
-            "gt": pc.greater,
-            "le": pc.less_equal,
-            "ge": pc.greater_equal,
-        }
+        # pyarrow 0.16.0 adds a compute module (thus the above compute import
+        # will work) but the attributes below are not available until pyarrow
+        # 1.0.0 resulting in an Attribute Error with pyarrow (0.16.0, 1.0.0].
+        try:
+            ARROW_CMP_FUNCS = {
+                "eq": pc.equal,
+                "ne": pc.not_equal,
+                "lt": pc.less,
+                "gt": pc.greater,
+                "le": pc.less_equal,
+                "ge": pc.greater_equal,
+            }
+        except AttributeError:
+            pass
```

Comment:
you need to define `ARROW_CMP_FUNCS = {}` (or rather do that uptop).

Reply:
It appears as such but I don't think it matters since you can't have pyarrow backed StringArrays unless you're on 1.0.0. I'm happy to add it though, I just noted that the original implementation didn't set to an empty dict either 9f `pyarrow.compute` was not available

Reply:
Looking at it more, the use `ARROW_CMP_FUNCS` assumes that it is populated with functions and thus I would have to make more changes -- but I feel like that might be outside of the scope of this PR for the reason I said above (considering the code won't be used unless `1.0.0`)